cmd/new.go

```diff
@@ -112,12 +116,165 @@ func scaffoldRule(params newRuleCommandParams) error {
 	}
 
 	if params.type_ == "builtin" {
+		if err := addToDataYAML(params); err != nil {
+			return err
+		}
+
+		if err := addRuleToREADME(params); err != nil {
+			return err
+		}
+
 		return scaffoldBuiltinRule(params)
 	}
 
 	return fmt.Errorf("unsupported type %v", params.type_)
 }
 
+func addToDataYAML(params newRuleCommandParams) error {
+	// Read the YAML content from the file
+	yamlContent, err := os.ReadFile("bundle/regal/config/provided/data.yaml")
+	if err != nil {
+		return err
+	}
+
+	var existingConfig config.Config
+
+	// Unmarshal the YAML content into a map
+	if err := yaml.Unmarshal(yamlContent, &existingConfig); err != nil {
+		return err
+	}
+
+	// Add the new entry to the Rules map within the Config struct
+	if existingConfig.Rules == nil {
+		existingConfig.Rules = make(map[string]config.Category)
+	}
+
+	// Create a new Rule value and set the Level field
+	vrule := config.Rule{
+		Level: "error",
+	}
+
+	// Check if the category already exists in rules
+	if existingConfig.Rules[params.category] == nil {
+		existingConfig.Rules[params.category] = make(map[string]config.Rule)
+	}
+
+	// Assign the new Rule value to the Category map
+	existingConfig.Rules[params.category][params.name] = vrule
+
+	// Sort the map keys alphabetically (categories)
+	sortedCategories := make([]string, 0, len(existingConfig.Rules))
+	for cat := range existingConfig.Rules {
+		sortedCategories = append(sortedCategories, cat)
+	}
+
+	sort.Strings(sortedCategories)
+
+	// Sort rule names within each category alphabetically
+	for _, cat := range sortedCategories {
+		sortedRuleNames := make([]string, 0, len(existingConfig.Rules[cat]))
+		for ruleName := range existingConfig.Rules[cat] {
+			sortedRuleNames = append(sortedRuleNames, ruleName)
+		}
+
+		sort.Strings(sortedRuleNames)
+
+		sortedCategory := make(config.Category)
+		for _, ruleName := range sortedRuleNames {
+			sortedCategory[ruleName] = existingConfig.Rules[cat][ruleName]
+		}
+
+		existingConfig.Rules[cat] = sortedCategory
+	}
+
+	var b bytes.Buffer
+
+	yamlEncoder := yaml.NewEncoder(&b)
+	yamlEncoder.SetIndent(2)
+
+	err = yamlEncoder.Encode(existingConfig)
+	if err != nil {
+		return err
+	}
+
+	// Write the YAML content to the file
+	return os.WriteFile("bundle/regal/config/provided/data.yaml", b.Bytes(), 0o600)
+}
+
+func addRuleToREADME(params newRuleCommandParams) error {
```

Comment:
This is clever! I wonder though, won't the "startPattern" break if we introduce a rule with a longer name than what we've previously had?

There's some code from before to render the table in the README over here: https://github.com/StyraInc/regal/blob/main/cmd/table.go

Do you think it would be worth refactoring that into a common function that we could reuse here?

Reply:
Thanks for the comments, I will investigate further.